### PR TITLE
[DNM] PyStein Trigger Templates

### PR DESCRIPTION
**Pull request by shreyabatra4**
Adding Http trigger template for Python new programming model

### CODE REVIEW

File: Functions.Templates/Templates/HttpTrigger-Python-Preview/function_app.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import azure.functions as func
+
+import logging
+
+app = func.FunctionApp(auth_level=func.AuthLevel.ANONYMOUS)
+
+@app.function_name(name="HttpTrigger1")
```
Comment (philliphoff) [inline Functions.Templates/Templates/HttpTrigger-Python-Preview/function_app.py:7]: As this template is intended to be appended to an existing `function_app.py`, I would think it should not include imports or the creation of the `app`.

Comment (shreyabatra4) [reply Functions.Templates/Templates/HttpTrigger-Python-Preview/function_app.py:1]: agreed, have removed

File: Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import logging
+
+import azure.functions as func
+
+app = func.FunctionApp()
+
+@app.function_name(name="mytimer")
+@app.schedule(schedule="0 */5 * * * *", arg_name="mytimer", run_on_startup=True,
+              use_monitor=False)
```
Comment (vrdmr) [inline Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md:22]:
```suggestion
@app.schedule(schedule="0 */5 * * * *", arg_name="mytimer") 
```

Simplifying this - Please remove function_nameand other optional fields.

File: Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import logging
+
+import azure.functions as func
+
+app = func.FunctionApp()
+
+@app.function_name(name="mytimer")
+@app.schedule(schedule="0 */5 * * * *", arg_name="mytimer", run_on_startup=True,
+              use_monitor=False) 
+def test_function(mytimer: func.TimerRequest) -> None:
```
Comment (vrdmr) [inline Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md:23]:
```suggestion
def mytimer(mytimer: func.TimerRequest) -> None:
```

File: Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md (hunk 2 above)
Comment (vrdmr) [reply Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md:22]: @shreyabatra4 any thoughts on this?

Comment (shreyabatra4) [reply Functions.Templates/Templates/TimerTrigger-Python-Preview/timer_trigger_template.md:22]: I feel like it's helpful to have function_name as well as other optional fields to demonstrate that they exist and can be used. Maybe I can add a comment stating they are optional to bring that clarity?

**Review by michaelpeng36 — COMMENTED** [review]
Would it be possible to add a keyword to the templates to replace with the user-supplied function name? For example, we could have the key word `FunctionNameToReplace`, so the template would start with `@app.function_name(name="FunctionNameToReplace")`, and later replace it with whatever the user specifies.

If we decide to move forward with this, then we'll just have to be careful not to use the key word in other contexts where we do not want the word replaced with a user-specified argument.

File: Functions.Templates/Templates/BlobTrigger-Python-v2-Append/blob_trigger_template.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+The Blob storage trigger starts a function when a new or updated blob is detected. The blob contents are provided as input to the function. The Azure Blob storage trigger requires a general-purpose storage account. Storage V2 accounts with hierarchical namespaces are also supported.
+
+## Using the Template
+
+Following is an example code snippet for Blob Trigger using the [Python programming model V2](https://aka.ms/pythonprogrammingmodel) (currently in Preview).
+
+```python
+import logging
+import azure.functions as func
+app = func.FunctionApp()
```
Comment (vrdmr) [inline Functions.Templates/Templates/BlobTrigger-Python-v2-Append/blob_trigger_template.md:14]: In this example and others below - please add linebreaks at the right places

File: Functions.Templates/Templates/BlobTrigger-Python-v2-Append/blob_trigger_template.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                f"Blob Size: {myblob.length} bytes")
+```
+
+To run the code snippet generated through the command palette, note the following:
+
+- The function application is defined and named `app`.
+- Confirm that the parameters within the trigger reflect values that correspond with your storage account.
+- The name of the file must be `function_app.py`.
+  
+Note that Blob input and output bindings are also supported in Azure Functions. To learn more, see [Azure Blob storage bindings overview](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob?tabs=in-process%2Cextensionv5%2Cextensionv3&pivots=programming-language-python)
```
Comment (vrdmr) [inline Functions.Templates/Templates/BlobTrigger-Python-v2-Append/blob_trigger_template.md:30]: Can you please use existing aka.ms links for this? This helps in tracking their success.

File: Functions.Templates/Templates/BlobTrigger-Python-v2/function_app.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import logging
+import azure.functions as func
+app = func.FunctionApp()
+@app.function_name(name="BlobTrigger1")
```
Comment (vrdmr) [inline Functions.Templates/Templates/BlobTrigger-Python-v2/function_app.py:4]: Please correct linebreaks in all the examples, templates, etc.

File: Functions.Templates/Templates/EventHubTrigger-Python-v2-Append/eventhub_trigger_template.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+Following is an example code snippet for Event Hub Trigger using the [Python programming model V2](https://aka.ms/pythonprogrammingmodel) (currently in Preview).
+
+```python
+import logging
+import azure.functions as func
+app = func.FunctionApp()
+@app.function_name(name="EventHubTrigger1")
+@app.event_hub_message_trigger(arg_name="myhub", event_hub_name="samples-workitems",
+                               connection=""CONNECTION_SETTING"")
```
Comment (vrdmr) [inline Functions.Templates/Templates/EventHubTrigger-Python-v2-Append/eventhub_trigger_template.md:17]: Check `""` issues. In all the places. Also in the above example, the variable was mentioned within `<` and `>` - please make the changes in all the places you use variables.

File: Functions.Templates/Templates/FunctionApp-Python-Preview/getting_started.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+If you have not already, please checkout our [quickstart](https://aka.ms/fxpythonquickstart) to get you started with Azure Functions developments in Python.
+
+## Publishing your function app to Azure
+  
+For more information on deployment options for Azure Functions, please visit this [guide](https://docs.microsoft.com/en-us/azure/azure-functions/create-first-function-vs-code-python#publish-the-project-to-azure).
+
+## Next Steps
+  
+To learn more specific guidance on developing Azure Functions with Python, please visit [Azure Functions Developer Python Guide](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python?tabs=asgi%2Capplication-level).
```
Comment (vrdmr) [inline Functions.Templates/Templates/FunctionApp-Python-Preview/getting_started.md:48]: some existing aka.ms link should come in

File: Functions.Templates/Templates/FunctionApp-Python-Preview/host.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+      "applicationInsights": {
+        "samplingSettings": {
+          "isEnabled": true,
+          "excludedTypes": "Request"
+        }
+      }
+    },
+    "extensionBundle": {
+      "id": "Microsoft.Azure.Functions.ExtensionBundle",
+      "version": "[3.15.0, 4.0.0)"
```
Comment (vrdmr) [inline Functions.Templates/Templates/FunctionApp-Python-Preview/host.json:13]: When are we moving to v4 extensions? Should this be updated in the future?

File: Functions.Templates/Templates/TimerTrigger-Python-v2-Append/function_app.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+@app.function_name(name="mytimer")
+@app.schedule(schedule="0 */5 * * * *", arg_name="mytimer", run_on_startup=True,
+              use_monitor=False) 
+def test_function(mytimer: func.TimerRequest) -> None:
```
Comment (vrdmr) [inline Functions.Templates/Templates/TimerTrigger-Python-v2-Append/function_app.py:4]: Add a comment here to tell customers to ensure `datetime` is imported.

**Comment by soninaren** [comment]
Closing this one as we have new templates that follow new spec.

**Pull request closed without merging**